web/src/model/filters.ts

```diff
@@ -3,17 +3,9 @@ import { isEqual } from '../utils/base-compare';
 
 export type FiltersEncoder = (values: FilterValue[], matchAny: boolean, not: boolean, moreThan: boolean) => string;
 
-export enum FilterComponent {
-  Autocomplete,
-  Text,
-  Number
-}
+export type FilterComponent = 'autocomplete' | 'text' | 'number';
 
-export enum FilterCategory {
-  Source,
-  Destination,
-  None
-}
+export type FilterCategory = 'source' | 'destination';
 
 export type TargetedFilterId =
```

Comment:
What happens if a user configures a filter with an id that isn't in that list? (e.g. imagine there's a new field we forgot to add here) - Is it still going to work?

Reply:
Nothing. The columns are still working fine since it's javascript behind the scene.
However, note that some behaviors are [still hardcoded by column id](https://github.com/netobserv/network-observability-console-plugin/blob/main/web/src/components/netflow-record/record-field.tsx#L238) as written in the description above.